openedx2zim/utils.py

```diff
@@ -351,18 +306,49 @@ def dl_dependencies(content, path, folder_name, c):
                 filename = sha256(str(src).encode("utf-8")).hexdigest() + ext
                 out = os.path.join(path, filename)
                 if not os.path.exists(out):
-                    download(unquote(src), out, c.conf["instance_url"])
+                    scraper.download_file(
+                        prepare_url(unquote(src), c.conf["instance_url"]),
+                        pathlib.Path(out),
+                    )
                 src = os.path.join(folder_name, filename)
                 iframe.attrib["src"] = src
     if imgs or docs or csss or jss or sources or iframes:
         content = html2string(body, encoding="unicode")
     return content
 
 
-def make_dir(path):
-    if not os.path.exists(path):
-        os.makedirs(path)
-
-
 def first_word(text):
     return " ".join(text.split(" ")[0:5])
+
+
+def get_meta_from_url(url):
+    def get_response_headers(url):
+        for attempt in range(5):
+            try:
+                return requests.head(url=url, allow_redirects=True, timeout=30).headers
+            except requests.exceptions.Timeout:
+                logger.error(f"{url} > HEAD request timed out ({attempt})")
+        raise Exception("Max retries exceeded")
+
+    try:
+        response_headers = get_response_headers(url)
+    except Exception as exc:
+        logger.error(f"{url} > Problem with head request\n{exc}\n")
+        return None, None
+    else:
+        filetype = response_headers.get("content-type", None)
```

Comment:
Please use `content_type` as to no create a third name for the already subtle mime/content-type.

Reply:
Done